### DISCUSSION TITLE
Build errors using GHC 9.8 due to missing (GHC-internal) library `system-cxx-std-lib`

### DISCUSSION THREAD

**Discussion by raehik** [srid/haskell-flake#200] [Q&A] [answered]
Upon trying to use `basePackages = pkgs.haskell.packages.ghc98` (on `haskell-updates`, and the branch over here https://github.com/NixOS/nixpkgs/pull/260138 ), I get the following error:

```
error:
       … while evaluating the attribute 'optionalValue.value'

(snip)

       error: function 'anonymous lambda' called without required argument 'system-cxx-std-lib'

       at /nix/store/q5p1xzzh61gb91k5y3zy3jj6w956frpc-source/pkgs/development/haskell-modules/hackage-packages.nix:1540:6:

         1539|   "BesselJ" = callPackage
         1540|     ({ mkDerivation, base, gamma, numerical-integration
             |      ^
         1541|      , system-cxx-std-lib, tasty, tasty-hunit
```

On a quick glance, this is a GHC thing, which may have existed since GHC 9.4. https://github.com/input-output-hk/haskell.nix/issues/1939

The nixpkgs branch I linked above has a comment stating that building the compiler/pkgset worked. Is there something in haskell-flake that's breaking the build?

Testing here: https://github.com/raehik/generic-data-functions/pull/3

**Comment by srid** [comment, marked as answer]
> ` error: function 'anonymous lambda' called without required argument 'system-cxx-std-lib'`

Errors like this typically mean that the specified package is not in the package set. That is to say, there is no `pkgs.haskell.packages.ghc98.system-cxx-std-lib`.

haskell-flake works with your `basePackages` as-is, except for merging your `packages`/`settings` as overlays; see https://github.com/srid/haskell-flake/blob/63a160879e6700b1c9385852043b162183063e9a/nix/modules/project/outputs.nix#L86-L91

**Reply by raehik** [reply]
I made this issue assuming the package set was configured properly, but I think I might have been wrong. I added `system-cxx-std-lib = null;` to `configuration-ghc-9.8.x.nix`, and it seems to be building fine. Thanks for the help. (Also sorry I tend to make issues over discussions, habit -- I'll remember for next time!)
